discord/ext/commands/core.py

```diff
@@ -715,18 +715,14 @@ async def call_before_hooks(self, ctx):
         # first, call the command local hook:
         cog = self.cog
         if self._before_invoke is not None:
-            try:
-                instance = self._before_invoke.__self__
-                # should be cog if @commands.before_invoke is used
-            except AttributeError:
-                # __self__ only exists for methods, not functions
-                # however, if @command.before_invoke is used, it will be a function
-                if self.cog:
-                    await self._before_invoke(cog, ctx)
-                else:
-                    await self._before_invoke(ctx)
-            else:
+            # should be cog if @commands.before_invoke is used
+            instance = getattr(self._before_invoke, '__self__', self.cog)
```

Comment:
```suggestion
            instance = getattr(self._before_invoke, '__self__', cog)
```

Defined above.

Also this probably needs to apply to `after_invoke` as well.

Reply:
Both suggested changges have been made.